### ISSUE TITLE
pwsh/powershell VT quirk causes conhost crash when resizing while alt buffer is being used

### ISSUE THREAD

**Issue by DHowett** [microsoft/terminal#13037]
Repro:

```
while($true){ "`e[?1049h";sleep 0.25;"`e[?1049l"; sleep 0.25}
```

Resize the window horizontally while it's running.

Toggling the alt buffer _from pwsh.exe or powershell.exe_ (specifically) causes us to UAF the outgoing `SCREEN_INFORMATION` here:
https://github.com/microsoft/terminal/blob/71cbdc8a1f6b8da10d130c053489445e99a4fc13/src/host/_stream.cpp#L1038-L1040

**Comment by zadjii-msft** [comment]
note: We greatly reduced the number of places where this can trigger in #13352, but this quirk isn't entirely gone, so this bug persists.

**Comment by DHowett** [comment]
Whoa, thanks for this! A repro that doesn't involve the powershell quirk (which we just removed!) makes this much more important, too.

**Comment by zadjii-msft** [comment]
Okay, coming back to this thread now that I'm back from leave and vacation. 

I can't seem to repro this with weechat in `main` anymore. There've been some locking and lifetime changes over the last few months that might have fixed this. @shuffle2 if you can't repro this on `main` anymore, then I suggest we punt this out to the backlog. The set of folks with the busted PsReadline that we still need to quirk for who are resizing their terminal while entering the alt buffer is gonna be samll and only shrinking. We can elevate from the backlog if we see live hits here, but I don't think there are any.

Actually heck, @shuffle2's crash is a different thing anyways. The pwsh VT quirk only applies to conhost, and that stack is firmly in _terminal_ code. I'm gonna punt and assume this was fixed. I vaguely recall some other resizing crash fixes, but 1.17 has been like, 4 months now.
